### ISSUE TITLE
Reconcile FormFieldMessage propTypes and typescript FormPropTypes.message

### ISSUE THREAD

**Issue by eschiebel** [instructure/instructure-ui#815]
### Context

FormPropTypes.message and FormFieldMessage

### Summary

1. **reconcile types**
From FormFieldMessage
  ```
static propTypes = {
    variant: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only']),
    children: PropTypes.node
  }
```
From FormPropTypes.message
```
const FormPropTypes = {
  message: PropTypes.shape({
    text: PropTypes.string,
    type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
  })
}
```
Update FormPropTypes.message to permit a string or a node
 
2. **Support `type='warning'`**

### Justification

1. Fix inconsistency
2. Include `type=warning` for completeness (and I could use it in a feature)

**Comment by HerrTopi** [comment]
Thanks for the detailed report, we will look into it!

**Comment by eschiebel** [comment]
Reconciling the types is good.
Adding the warning type might have rippling affects, like the state of various inputs when given a warning message.
Not sure what we want to do here. I've been able to work around it by using a "hint" message but coloring it "warning".

**Comment by HerrTopi** [comment]
Yeah, we have to figure out the warning, for sure. We'll have an internal meeting about this soon and we'll figure it out (hopefully :D )

**Comment by ke1k0** [comment]
Hi @eschiebel ! I'm already working on making the messages accept nodes as well.

I discussed the second half of your request with the design team, and they told me that it is not possible/allowed to add a 'warning' style message to the FormFields.

The reason behind it is that the 'orange' color used for warning (Fire) doesn't meet the color contrast requirements when used as text (it is only 3.11:1 with white background, and that is not sufficient for WCAG AA).

Ákos suggested to use the "hint" variant, or if it is a strong warning, then the "error". Also, after this feature messages will be able to receive Nodes, so another possible solution is to combine a "warning" icon with a "hint" type text message.

In case you need more info on this, let me know and send us the design plan of the feature or contact the design team directly! :)

**Comment by ke1k0** [comment]
PR-s for the propType issue:

V8: https://github.com/instructure/instructure-ui/pull/819
V7: https://github.com/instructure/instructure-ui/pull/820

**Comment by ke1k0** [comment]
Hi @eschiebel !
I just wanted to let you know that the PR-s for both V8 and V7 have been merged and will be shipped with the next release.
Also, has the information I provided about the warning messages been useful or do you need further info/help?

**Comment by ke1k0** [comment]
Hi @eschiebel ! 

This feature has been released for both [V8](https://instructure.design/#CHANGELOG) and [V7](https://instructure.design/v7/#CHANGELOG).

Do you have further questions about the warning messages, or can I close this ticket?

**Comment by eschiebel** [comment]
@ke1k0 you note that
> The reason behind it is that the 'orange' color used for warning (Fire) doesn't meet the color contrast requirements when used as text (it is only 3.11:1 with white background, and that is not sufficient for WCAG AA).

yet `<Text color="warning">` is valid. Doesn't seem right. Shouldn't the recent color update have taken care of this?

**Comment by ke1k0** [comment]
Yes, we are planning to remove the `warning` color version of `Text` as well, because it doesn't have the sufficient contrast ratio with the background. We are still in the process of resolving these color contrast issues for the new requirements :)

Also, the warning (Fire) color wasn't modified in the last color update, since modifying it would result in a color almost identical to the error color (Crimson). So the design team decided to leave it as it is and only use it for decoration and presentation purposes, but it is not allowed to use as text color.
